src/inventory.cpp

```diff
@@ -271,6 +271,30 @@ std::string ItemStack::getShortDescription(const IItemDefManager *itemdef) const
 	return desc;
 }
 
+f32 ItemStack::getRange(const IItemDefManager *itemdef) const
+{
+    std::string sRange = metadata.getString("range");
+    f32 range = -1;
+    if (sRange.empty())
+        range = getDefinition(itemdef).range;
+    if (!sRange.empty())
+    {
+        try {
```

Comment:
`std::strtof` doesn't throw on parse failure; you don't need this. `std::stof` does throw, however.